okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/QosExceptionResponseHandler.java

```diff
@@ -39,16 +39,22 @@
 
     @Override
     public Optional<QosException> handle(Response response) {
+        final Optional<QosException> result;
         switch (response.code()) {
             case 308:
-                return handle308(response);
+                result = handle308(response);
+                break;
             case 429:
-                return Optional.of(handle429(response));
+                result = Optional.of(handle429(response));
+                break;
             case 503:
-                return Optional.of(handle503());
+                result = Optional.of(handle503());
+                break;
+            default:
+                result = Optional.empty();
         }
-
-        return Optional.empty();
+        result.ifPresent(exception -> response.close());
```

Comment:
looks like we have to do this in all code paths that consumer the response.

Reply:
the other ones all try-with-resources it, so it's unnecessary.